src/apps/integrations/service/integrations.py

```diff
@@ -125,4 +125,23 @@ async def retrieve_integration(self, applet_id, integration_type) -> Integration
                     configuration=future_integration,
                 )
             case _:
-                raise UnsupportedIntegrationError(type=integration_type)
+                raise UnsupportedIntegrationError(integration_type=integration_type)
+
+    async def delete_integration(self, applet_id, integration_type) -> Integration:
```

Comment:
This could be renamed `delete_integration_by_type`.

I think `delete_integration` should receive the `integration_id` and call that directly instead without having to retrieve first.